### PR TITLE
boards/z1: fix broken clock configuration

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

The MSP430F2xx family has on RSEL bit more than the MSP430x1xxx family. The first commit updates the clock calibration accordingly.

https://github.com/RIOT-OS/RIOT/commit/df5c319978570f90071bbc83ddf10eea5449d46d from https://github.com/RIOT-OS/RIOT/pull/19558 broke the clock configuration of the Z1 by relying on the incorrect documentation of what clock is actually used. Closely reading the convoluted clock initialization code revealed that no XT2 crystal is present (as also indicated by some comments in `board.c`), contradicting the `#define MSP430_HAS_EXTERNAL_CRYSTAL 1` in the `board.h`.

The second commit should restore behavior (but with calibrated DCO than hard coded magic numbers).

### Testing procedure

`make BOARD=z1 -C examples/default flash term`

should work again, but fail with `master`.

### Issues/PRs references

Regression introduced in https://github.com/RIOT-OS/RIOT/pull/19558

### CODE REVIEW

**Comment by maribu** [comment]
@kaspar030 tested this PR, see https://gitea.riot-labs.de/kaspar/results/src/branch/main/pr19705

The takeaway is:

- The board is starting again
- There still is lot of cleanup and bug fixing left to get MSP430 support back up to standard

**Comment by maribu** [comment]
I changed my mind regarding skipping the most significant RSEL bit in the calibration when an external resistor is used in the DCO: Just doing the normal calibration anyway reduces code complexity and size and only adds a minor delay in boot up time.

File: cpu/msp430fxyz/clock.c [hunk 1]
```diff
@@ -188,7 +194,13 @@ static void calibrate_dco(void)
     uint8_t bcsctl1 = BCSCTL1 & ~(RSEL_MASK);
     uint8_t rselx = 0;
 
-    for (uint8_t iter = 0x04; iter != 0; iter >>= 1) {
+    /*
+     * Note: For MSP430F2xx (HAS_RSEL3 == 1) the bit in RSEL3 is actually
+     * ignored if an external resistor is used for the DCO. Still, settting
```
Comment (benpicco) [inline cpu/msp430fxyz/clock.c:199]:
```suggestion
     * ignored if an external resistor is used for the DCO. Still, setting
```

**Review by benpicco — APPROVED** [review]
Code looks good and @kaspar030 confirmed it's still working

**Comment by maribu** [comment]
bors merge

**Comment by maribu** [comment]
Thx :)

**Pull request merged**